pkg/compose/v1/app_store.go

```diff
@@ -88,7 +88,10 @@ func (s *appStore) RemoveApps(ctx context.Context, apps []*compose.AppRef, prune
 	for _, a := range apps {
 		appVerDir := filepath.Join(s.appsRoot, a.Name, a.Digest.Encoded())
 		if _, err := os.Stat(appVerDir); os.IsNotExist(err) {
-			return fmt.Errorf("app dir does not exist: %s", appVerDir)
+			// TODO: add debug level logging instead of just printing message unconditionally
+			fmt.Printf("app dir does not exist in the store's app dir,"+
+				" will check and remove app blobs only from the store's blob dir: %s", appVerDir)
+			continue
```

Comment:
I think a safer way to do this is to try and remove the directory and catch this error there. Otherwise, its still theoretically possible for this to fail in this same way.

Reply:
`RemoveAll` doesn't return an error if a path doesn't exist, and the implementation needs to know whether the dir exists or not.